File: 120. Triangle-2.java
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        int ans = Integer.MAX_VALUE;
        int n = triangle.size();
        for(int i = n - 2; i >= 0; i--){
            int m = triangle.get(i).size();
            for(int j = 0; j < m; j++){
                int left = triangle.get(i + 1).get(j);
                int right = triangle.get(i + 1).get(j + 1);
                int curr = triangle.get(i).get(j);
                triangle.get(i).set(j, Math.min(curr + left, curr + right));
            }
        }

        return triangle.get(0).get(0);
    }
}

File: 120. Triangle-2.cpp
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        int n = triangle.size();
        for (int i = n - 2; i >= 0; i--) {
            int m = triangle[i].size();
            for (int j = 0; j < m; j++) {
                int left = triangle[i + 1][j];
                int right = triangle[i + 1][j + 1];
                int curr = triangle[i][j];
                triangle[i][j] = min(curr + left, curr + right);
            }
        }
        return triangle[0][0];
    }
};
